include/swift/AST/DiagnosticsSema.def

```diff
@@ -1735,7 +1735,10 @@ ERROR(functions_mutating_and_not,none,
       "method must not be declared both %0 and %1",
       (SelfAccessKind, SelfAccessKind))
 ERROR(static_functions_not_mutating,none,
-      "static functions must not be declared mutating", ())
+      "%0 modifier cannot be used on static functions",
+      (SelfAccessKind))
```

Comment:
If you incorporate the declaration like so, the message will be also be tailored for an accessor:
```suggestion
      "%0 modifier cannot be used on %kindonly1",
      (SelfAccessKind, const FuncDecl *))
```

We loose the plural, but I think that’s fine because the object remains indefinite.